### PR TITLE
Add palette summary to user guide

### PR DESCRIPTION
**Pull request by canavandl**
ref issues #1873, #2218 

Added a description of palettes to the User Guide. Let me know if I need to change anything, I'm still trying to get the "voice" of your docs.

### CODE REVIEW

File: sphinx/source/docs/user_guide/styling.rst [hunk 1]
```diff
@@ -20,6 +20,29 @@ following ways:
...
+
+.. code-block:: python
+
+    >>> from bokeh.palettes import Spectral6
+    >>> print(Spectral6)
+    ['#3288bd', '#99d594', '#e6f598', '#fee08b', '#fc8d59', '#d53e4f']
+
+All of the standard palettes included in bokeh can be found at
+:ref:`bokeh_dot_palettes`. Custom palettes can be created by instantiating
+arbitrary sequences of RGB(A) hex strings.
```
Comment (bryevdv) [inline sphinx/source/docs/user_guide/styling.rst]: "creating sequences" a little less verbose

File: sphinx/source/docs/user_guide/styling.rst [hunk 2]
```diff
@@ -20,6 +20,29 @@ following ways:
...
+``bokeh.charts`` and as the ``color`` attribute of many plot objects from
+``bokeh.plotting``. Bokeh offers many of the standard Brewer palettes, which
+can be imported from the ``bokeh.palettes`` module. For example, importing
+“Spectral6” gives a linearly-spaced, six element list of RBG(A) hex strings
+from the Brewer “Spectral” colormap.
+
+.. code-block:: python
+
+    >>> from bokeh.palettes import Spectral6
+    >>> print(Spectral6)
```
Comment (bryevdv) [inline sphinx/source/docs/user_guide/styling.rst]: maybe just `Spectral16` and not `print(Spectral16)`? The repr would look the same but not rely on python 3 print syntax

File: sphinx/source/docs/user_guide/styling.rst [hunk 3]
```diff
@@ -20,6 +20,29 @@ following ways:
...
 
+Using Palettes
+-----------------
+
+Palettes are sequences (lists or tuples) of RGB(A) hex strings that define a
+colormap and can set as the ``palette`` attribute of all chart types from
+``bokeh.charts`` and as the ``color`` attribute of many plot objects from
+``bokeh.plotting``. Bokeh offers many of the standard Brewer palettes, which
+can be imported from the ``bokeh.palettes`` module. For example, importing
+“Spectral6” gives a linearly-spaced, six element list of RBG(A) hex strings
```
Comment (bryevdv) [inline sphinx/source/docs/user_guide/styling.rst]: "linearly-spaced" refers to the action the color-mapper, which linearly maps scalar values between a fixed range into the color bins. I would just say "provides a six element list..."

File: sphinx/source/docs/user_guide/styling.rst [hunk 4]
```diff
@@ -20,6 +20,29 @@ following ways:
 
 .. _userguide_styling_visual_properties:
```
Comment (bryevdv) [inline sphinx/source/docs/user_guide/styling.rst]: I think the ref labels are swapped (this one with the one below in the next section)

**Comment by bryevdv** [comment]
A few small comments, otherwise looks great

**Comment by canavandl** [comment]
I added your recommendations. Thanks @bryevdv

**Comment by bryevdv** [comment]
LGTM!

**Comment by bryevdv** [comment]
Dunno what's up with a merge conflict tho?

**Comment by canavandl** [comment]
I think it's because I moved the :ref: label around. I'll deal w/ it in the AM

**Comment by canavandl** [comment]
nvm - somebody merged some Scala docs to the user guide and I was behind on master. I'm good now.

**Pull request merged**